src/test/java/org/jsoup/parser/HtmlTreeBuilderStateTest.java

```diff
@@ -1,38 +1,27 @@
 package org.jsoup.parser;
 
-import org.jsoup.parser.HtmlTreeBuilderState.Constants;
 import org.junit.Test;
 
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
 
 import static org.junit.Assert.assertArrayEquals;
 
 public class HtmlTreeBuilderStateTest {
     @Test
-    public void ensureArraysAreSorted() {
-        String[][] arrays = {
-            Constants.InBodyStartToHead,
-            Constants.InBodyStartPClosers,
-            Constants.Headings,
-            Constants.InBodyStartPreListing,
-            Constants.InBodyStartLiBreakers,
-            Constants.DdDt,
-            Constants.Formatters,
-            Constants.InBodyStartApplets,
-            Constants.InBodyStartEmptyFormatters,
-            Constants.InBodyStartMedia,
-            Constants.InBodyStartInputAttribs,
-            Constants.InBodyStartOptions,
-            Constants.InBodyStartRuby,
-            Constants.InBodyStartDrop,
-            Constants.InBodyEndClosers,
-            Constants.InBodyEndAdoptionFormatters,
-            Constants.InBodyEndTableFosters,
-            Constants.InCellNames,
-            Constants.InCellBody,
-            Constants.InCellTable,
-            Constants.InCellCol,
-        };
+    public void ensureArraysAreSorted() throws ClassNotFoundException, IllegalAccessException {
+        Class constants = Class.forName("org.jsoup.parser.HtmlTreeBuilderState$Constants");
+        Field[] fields = constants.getDeclaredFields();
+        List<String[]> arrays = new ArrayList<>(fields.length);
+        for (Field field : fields) {
+            int modifiers = field.getModifiers();
+            if (Modifier.isStatic(modifiers) && Modifier.isFinal(modifiers) && field.getType() == String[].class) {
```

Comment:
My main concern with changing the test to use reflection like this is that the test could, now or in the future, accidentally be skipping fields that need to be tested. E.g. what if down the line we add a constant and for some reason don't mark it final? (I know a bit of a forced example.)

I think if we also add an explicit test that there were _n_ arrays matched it would be more explicit, and can keep a warning in the source to review the test.

Yes, would be good to update the other tests like this, and share a common piece of reflection code.